### PR TITLE
Fix #4209 - send the mail queue using a task that runs every minute

### LINKED ISSUE #4209: Move mail queue processing to a task rather than a shutdown function
The mail queue is currently ran using a shutdown function, which is registered [in `init.php`](https://github.com/mybb/mybb/blob/aacebaecb49d0cb951096134277250192a2cedba/inc/init.php#L243).

This can be problematic when using SMTP to send mail and the SMTP server is either slow to respond or offline, as it leads to slow page load times.

We should instead use a task to send the mail queue.

### PR DESCRIPTION
**Pull request by euantorano**
Fixes #4209.

This adds a new task, `sendmailqueue`, which defaults to running every minute.

It also adds a new setting to manage the maximum number of messages to send from the mail queue when the task runs - previously only the default value of 10 emails would be used.

### CODE REVIEW

**Review by Ben-MyBB — APPROVED** [review]
Aside from the conflicts on this PR it all looks good. @mybb/developers given that I'm not overly experienced at the minute could someone else give this a look over please?

**Comment by euantorano** [comment]
I’ll sort the conflicts this evening.

**Comment by euantorano** [comment]
Conflicts resolved.

**Comment by Eldenroot** [comment]
I have applied to my board, seems to be fine, if anything I will add a note here.

File: install/resources/upgrade52.php [hunk 1]
```diff
@@ -42,4 +42,8 @@ function upgrade51_dbchanges()
 			$db->add_column("attachtypes", "forcedownload", "tinyint(1) NOT NULL default '0' AFTER enabled");
 			break;
 	}
+  
+	$added_tasks = sync_tasks();
+
+	echo "<p>Added {$added_tasks} new tasks.</p>";
```
Comment (Sama34) [inline install/resources/upgrade52.php:48]:
```suggestion
	echo "<p>Added {$added_tasks} new tasks.</p>";

	$output->print_contents("<p>Click next to continue with the upgrade process.</p>");
	$output->print_footer("52_done");
```

**Review by Sama34 — APPROVED** [review]
I think you should add the suggested code to the upgrade script.

File: install/resources/upgrade52.php (hunk 1 above)
Comment (Sama34) [reply install/resources/upgrade52.php:48]: @euantorano I did test this change, so you could either accept it or left it for later and merge.

**Pull request merged**